Lib/test/test_opcache.py

```diff
@@ -1362,6 +1362,49 @@ def binary_op_add_extend():
         self.assert_specialized(binary_op_add_extend, "BINARY_OP_EXTEND")
         self.assert_no_opcode(binary_op_add_extend, "BINARY_OP")
 
+        def binary_op_zero_division():
+            def compactlong_lhs(arg):
+                42 / arg
+            def float_lhs(arg):
+                42.0 / arg
+
+            with self.assertRaises(ZeroDivisionError):
+                compactlong_lhs(0)
+            with self.assertRaises(ZeroDivisionError):
+                compactlong_lhs(0.0)
+            with self.assertRaises(ZeroDivisionError):
+                float_lhs(0.0)
+            with self.assertRaises(ZeroDivisionError):
+                float_lhs(0)
+
+            self.assert_no_opcode(compactlong_lhs, "BINARY_OP_EXTEND")
+            self.assert_no_opcode(float_lhs, "BINARY_OP_EXTEND")
+
+        binary_op_zero_division()
+
+        def binary_op_nan():
+            def compactlong_lhs(arg):
+                42 + arg
+                42 - arg
+                42 * arg
+                42 / arg
+            def compactlong_rhs(arg):
+                arg + 42
+                arg - 42
+                arg * 42
+                arg / 42
+
+            compactlong_lhs(1.0)
+            for _ in range(100):
+                compactlong_lhs(float('nan'))
```

Comment:
While we're here, we might as well assert that the value returned from the function is correct.